src/sparseml/modifiers/utils/pytorch_helpers.py

```diff
@@ -58,9 +59,9 @@ def run_calibration_forward(
     )
 
     # run through the calibration data
-    num_iterations = num_calibration_steps or len(_dataloader)
-    for batch_idx in tqdm(range(num_iterations)):
-        batch = _dataloader[batch_idx]
+    for batch_idx, batch in enumerate(tqdm(_dataloader)):
+        if num_calibration_steps and batch_idx >= num_calibration_steps:
```

Comment:
Could we update the docstring to say that -negative num_calibration_steps will also process all data?